frontend/web/components/pages/HomePage.js

```diff
@@ -221,7 +221,10 @@ const HomePage = class extends React.Component {
         )
       }
 
-      if (Utils.getFlagsmithHasFeature('saml')) {
+      if (
+        !Utils.flagsmithFeatureExists('saml') ||
```

Comment:
Hmm... maybe this is fine, but I guess this will enable the 'Single Sign-On' button (admittedly it won't work) for pretty much all self hosted customers that _have_ set up Flagsmith on Flagsmith (I'm intrigued to know how many this actually is). 

I'm a little concerned that the experience for any curious open source self hosters will be pretty bad as well, because I don't think the endpoint will even exist for the call made when adding in the organisation name via 'Single Sign-On'.

Reply:
This is the current experience:

* Out of the box, "Single Sign-On" is shown, since all self-hosted images connect to SaaS FoF by default, which has the `saml` flag enabled
* If you set up FoF with no flags, "Single Sign-On" disappears until you create and enable the `saml` flag

After this change:

* Out of the box, "Single Sign-On" is shown as before
* If you set up FoF with no flags, "Single Sign-On" is still shown until you create and disable the `saml` flag

In any case, the error that you get when trying to use SSO is "Please check your organisation name and try again." despite the 405 response, so I'm not too worried about this. We could use the SSO button as an upsell opportunity, the same way we do now for features that are gated behind plans. Even without adding an explicit upsell action item, it still advertises that SSO exists is good enough without being confusing IMO.